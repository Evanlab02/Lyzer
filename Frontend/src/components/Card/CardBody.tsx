import { PropsWithChildren } from "react";

export default function CardBody(props: PropsWithChildren) {
	const { children } = props;

	return (
		<h3 className="lyzer-card-body">
			{children}
		</h3>
	);
}