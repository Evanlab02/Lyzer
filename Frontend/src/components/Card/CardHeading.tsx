import { PropsWithChildren } from "react";

export default function CardHeading(props: PropsWithChildren) {
    const { children } = props;

    return (
        <span className="lyzer-card-heading">
            {children}
        </span>
    )
}