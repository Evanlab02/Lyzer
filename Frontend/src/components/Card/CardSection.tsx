import { PropsWithChildren } from "react";

export interface CardSectionProps extends PropsWithChildren {
    title: string;
    subtitle: string;

}

export default function CardSection(props: Readonly<CardSectionProps>) {
	const { title, subtitle, children } = props;

	return (
		<div className="lyzer-card-section">
			<h3 className="lyzer-card-section-title">{title}</h3>
			<p className="lyzer-card-section-subtitle">{subtitle}</p>
			{children}
			<hr className="lyzer-card-section-divider" />
		</div>
	);
}