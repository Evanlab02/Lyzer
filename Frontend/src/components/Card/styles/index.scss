.lyzer-card {
    background-color: var(--card-bg-colour);
    border-radius: 8px;
    border: 1px solid var(--card-border-colour);
    padding: 16px;
    height: 100%;
    box-sizing: border-box;

    .lyzer-card-heading {
        font-weight: 700 !important;
        text-transform: uppercase !important;
        font-size: 12px !important;
    }

    .lyzer-card-body {
        margin: 10px 0 10px 0;
        font-size: 1.35rem;
        min-height: 42px;
    }
}