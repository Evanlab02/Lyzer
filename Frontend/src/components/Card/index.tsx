import CardBody from "./CardBody";
import CardHeading from "./CardHeading";
import { PropsWithChildren } from "react";
import "./styles/index.scss";

export interface CardProps extends PropsWithChildren {
    testId?: string;
}

export default function Card(props: CardProps) {
	const { children, testId } = props;

	return(
		<div className="lyzer-card" data-testid={testId}>
			{children}
		</div>
	);
}

export { CardBody, CardHeading };
