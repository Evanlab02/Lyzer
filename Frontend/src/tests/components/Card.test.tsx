import { expect, it } from "vitest";
import { render } from "@testing-library/react";
import Card, { CardBody, CardHeading } from "../../components/Card";

it("Basic card renders correctly", () => {
	const { getByTestId } = render(
		<Card testId="lyzer-card">
			Basic Card
		</Card>
	);

	const card = getByTestId("lyzer-card");
	expect(card).toMatchSnapshot();
});

it("Card with heading renders correctly", () => {
	const { getByTestId } = render(
		<Card testId="lyzer-card">
			<CardHeading>
                Card Heading
			</CardHeading>
		</Card>
	);

	const card = getByTestId("lyzer-card");
	expect(card).toMatchSnapshot();
});

it("Card with body renders correctly", () => {
	const { getByTestId } = render(
		<Card testId="lyzer-card">
			<CardBody>
                Card Body
			</CardBody>
		</Card>
	);

	const card = getByTestId("lyzer-card");
	expect(card).toMatchSnapshot();
});

it("Card with heading and body renders correctly", () => {
	const { getByTestId } = render(
		<Card testId="lyzer-card">
			<CardHeading>
                Card Heading
			</CardHeading>
			<CardBody>
                    Card Body
			</CardBody>
		</Card>
	);

	const card = getByTestId("lyzer-card");
	expect(card).toMatchSnapshot();
});
