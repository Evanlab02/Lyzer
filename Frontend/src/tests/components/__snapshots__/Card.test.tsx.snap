// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Basic card renders correctly 1`] = `
<div
  class="lyzer-card"
  data-testid="lyzer-card"
>
  Basic Card
</div>
`;

exports[`Card with body renders correctly 1`] = `
<div
  class="lyzer-card"
  data-testid="lyzer-card"
>
  <h3
    class="lyzer-card-body"
  >
    Card Body
  </h3>
</div>
`;

exports[`Card with heading and body renders correctly 1`] = `
<div
  class="lyzer-card"
  data-testid="lyzer-card"
>
  <span
    class="lyzer-card-heading"
  >
    Card Heading
  </span>
  <h3
    class="lyzer-card-body"
  >
    Card Body
  </h3>
</div>
`;

exports[`Card with heading renders correctly 1`] = `
<div
  class="lyzer-card"
  data-testid="lyzer-card"
>
  <span
    class="lyzer-card-heading"
  >
    Card Heading
  </span>
</div>
`;
